### PR TITLE
executor: buildWindow cannot call typeInfer twice

### PR DESCRIPTION
**Pull request by fzhedu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/30402

Problem Summary: build window desc call TypeInfer twice, which change the return type incorrectly. For example, avg() in window func, a int type is first inferred to decimal(15,4), then inferred to decimal(19,8), which breaks compatibility with mysql.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: expression/aggregation/descriptor.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ type AggFuncDesc struct {
 }
 
 // NewAggFuncDesc creates an aggregation function signature descriptor.
+// this func cannot be called twice as the TypeInfer will change return type.
```
Comment (XuHuaiyu) [inline expression/aggregation/descriptor.go:46]: Does this function change the return type for all the cases?

Comment (fzhedu) [reply expression/aggregation/descriptor.go:46]: [TypeInfer](https://github.com/pingcap/tidb/blob/155a6027db13e312e2b8d7d3b5a660a44e51f639/expression/aggregation/base_func.go#L87) change types especially for decimal type.

**Comment by fzhedu** [comment]
integration-common-test — Jenkins job failed.
result from index , none of business of this PR.

File: expression/aggregation/descriptor.go (hunk 1 above)
Comment (windtalker) [reply expression/aggregation/descriptor.go:46]: The comment is a little confusing, I think by `change return type` you actually mean `change the input type`?

Comment (fzhedu) [reply expression/aggregation/descriptor.go]: > The comment is a little confusing, I think by `change return type` you actually mean `change the input type`?

done

**Review by windtalker — APPROVED** [review]
LGTM

**Review by XuHuaiyu — APPROVED** [review]
LGTM

File: executor/builder.go [hunk 2]
```diff
@@ -4192,7 +4192,7 @@ func (b *executorBuilder) buildWindow(v *plannercore.PhysicalWindow) Executor {
 	partialResults := make([]aggfuncs.PartialResult, 0, len(v.WindowFuncDescs))
 	resultColIdx := v.Schema().Len() - len(v.WindowFuncDescs)
 	for _, desc := range v.WindowFuncDescs {
-		aggDesc, err := aggregation.NewAggFuncDesc(b.ctx, desc.Name, desc.Args, false)
```
Comment (winoros) [inline executor/builder.go:4195]: Why we call it twice?

Comment (fzhedu) [reply executor/builder.go:4195]: when doing the plan, the agg funcs have called TypeInfer, but here executing pahse, buildWindow do this again. In fact, we should not change the plan after final plan.

**Pull request merged**